src/HaroohiePals.Gui/ImGuiController.cs

```diff
@@ -281,36 +278,36 @@ private void RecreateFontDeviceTexture()
 
         io.Fonts.GetTexDataAsRGBA32(out IntPtr pixels, out int width, out int height, out int bytesPerPixel);
 
-        void loadIcon(int rectId, byte[] iconBytes)
+        foreach (var icon in _icons)
         {
-            using (var icon = Image.Load<Bgra32>(iconBytes))
-            {
-                unsafe
-                {
-                    uint* pixPtr = (uint*)pixels;
-                    var rect = io.Fonts.GetCustomRectByIndex(rectId);
-                    for (int y = 0; y < rect.Height; y++)
-                    {
-                        uint* p = pixPtr + (rect.Y + y) * width + (rect.X);
-                        for (int x = 0; x < rect.Width; x++)
-                            p[x] = icon[x, y].PackedValue;
-                    }
-                }
-            }
+            LoadIcon(registeredIcons[icon], pixels, icon.ImageBytes, width, height);
         }
 
-        foreach (var icon in _icons)
-            loadIcon(registeredIcons[icon], icon.ImageBytes);
-
         _fontTexture = new GLTexture(pixels, width, height, PixelFormat.Bgra, PixelType.UnsignedByte,
             TextureWrapMode.Clamp, TextureWrapMode.Clamp);
         _fontTexture.SetFilterMode(TextureFilterMode.Linear, TextureFilterMode.Linear);
 
         io.Fonts.SetTexID((IntPtr)_fontTexture.Handle);
-
         io.Fonts.ClearTexData();
     }
 
+    private unsafe void LoadIcon(int rectId, in IntPtr pixels, byte[] iconBytes, int width, int height)
```

Comment:
```suggestion
    private unsafe void LoadIcon(int rectId, IntPtr pixels, byte[] iconBytes, int width, int height)
```
IntPtr is just a pointer (e.g. a 64 bit number), so it doesn't need `in`.